sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/samples/Sample2_CNCF.md

```diff
@@ -0,0 +1,38 @@
+# Using the Cloud Native CloudEvent type
+
+It is possible to publish and receive the CloudNative CloudEvent type found in the [CloudNative.CloudEvents library](https://www.nuget.org/packages/CloudNative.CloudEvents) using the Azure Event Grid Namespaces client library.
+First we create a `CloudEvent` object and publish it to the namespace topic using the `EventGridSenderClient`.
+
+```C# Snippet:SendCNCFEvent
+var evt = new CloudNative.CloudEvents.CloudEvent
+{
+    Source = new Uri("http://localHost"),
+    Type = "type",
+    Data = new TestModel { Name = "Bob", Age = 18 },
+    Id = Recording.Random.NewGuid().ToString()
+};
+var jsonFormatter = new JsonEventFormatter();
+var sender = new EventGridSenderClient(new Uri(namespaceTopicHost), topicName, new AzureKeyCredential(namespaceKey));
+await sender.SendEventAsync(RequestContent.Create(jsonFormatter.EncodeStructuredModeMessage(evt, out _)));
+```
+
+Next, we receive the events using the `EventGridReceiverClient`.
+
+```C# Snippet:ReceiveCNCFEvent
+var receiver = new EventGridReceiverClient(new Uri(namespaceTopicHost), topicName, subscriptionName, new AzureKeyCredential(namespaceKey));
+Response response = await receiver.ReceiveAsync(maxEvents: 1, maxWaitTime: TimeSpan.FromSeconds(10), new RequestContext());
+using var document = JsonDocument.Parse(response.Content);
+
+var eventResponse = document.RootElement.GetProperty("value").EnumerateArray().First();
```

Comment:
I tried using ToDynamicJson instead, but it doesn't seem to allow you to let you convert to bytes, which I would need to pass to DecodeStructuredModeMessage.
e.g. 
I could do:
```c#
var evt = response.Content.ToDynamicJson.Value[0].Event;
```
But at that point I would need to somehow convert evt to bytes and I wasn't sure how to do this.

Reply:
Maybe something like:

```csharp
var byteStuff = Encoding.UTF8.GetBytes(
    response.Content.ToDynamicJson().Value[0].Event.ToString());
```

So long as `Event` is of type `string`, I believe that would work.

Reply:
Nice.. that works